spring-kafka/src/main/java/org/springframework/kafka/support/AbstractKafkaHeaderMapper.java

```diff
@@ -200,13 +206,58 @@ private String mapRawIn(String header, byte[] value) {
 	}
 
 
+	/**
+	 * A matcher for headers.
+	 * @since 2.3
+	 */
+	protected interface HeaderMatcher {
```

Comment:
So, the idea to let have any custom matcher only if we extend the whole class.
 No way to reuse default one, but insert custom matchers.

Thanks

Reply:
I simply realized that having a pattern matcher for header names without a wildcard is much less efficient than looking up the header in a set.

The interface is so I can have 2 different types of matcher.

Reply:
But my point that it is `protected` and therefore not available if we don't extend.
The implementation is fine though, only concern is about convenience of end-users.